kvbc/src/categorization/kv_blockchain.cpp

```diff
@@ -234,17 +441,25 @@ KeyValueUpdatesInfo KeyValueBlockchain::handleCategoryUpdates(
   return kvui;
 }
 
-ImmutableUpdatesInfo KeyValueBlockchain::handleCategoryUpdates(
-    BlockId block_id,
-    const std::string& category_id,
-    ImmutableUpdatesData&& updates,
-    concord::storage::rocksdb::NativeWriteBatch& write_batch) {
-  ImmutableUpdatesInfo skvui;
-  for (auto& [k, v] : updates.kv) {
-    (void)v;
-    skvui.tagged_keys[k] = std::move(v.tags);
+ImmutableUpdatesInfo KeyValueBlockchain::handleCategoryUpdates(BlockId block_id,
+                                                               const std::string& category_id,
+                                                               ImmutableUpdatesData&& updates,
+                                                               concord::storage::rocksdb::NativeWriteBatch& write_batch,
+                                                               categorization::RawBlock& raw_block) {
+  // if true means that category is new and we should create an instance of it.
+  if (insertCategoryMapping(category_id, detail::CATEGORY_TYPE::immutable, write_batch)) {
```

Comment:
Is it possible to first lookup in `categories_`, in order to avoid the DB lookup? Once we instantiate categories in the constructor, I suppose we can assume that the memory map is in sync with the DB and only update them both when a new category appears. Please let me know if I miss something.

Reply:
good catch, missed it, thanks!

Reply:
Thanks! I guess if we assume that we are creating categories rarely, we can even first look up in `categories_` (and not in `categories_types_`) and expect that most of the time that will be enough to know it is an existing category. Doing so can avoid the 2 lookups we need now (both in `categories_` and `categories_types_`).